src/components/Header/Header.jsx

```diff
@@ -0,0 +1,59 @@
+import React, { useState, useEffect } from 'react';
+import { usePrevious } from 'react-hanger';
+import PropTypes from 'prop-types';
+
+export const Header = ({ onAddNewTodo, todos }) => {
+  const [title, setTitle] = useState('');
+  const prevTodos = usePrevious(todos);
+
+  useEffect(() => {
+    const saveTitle = localStorage.getItem('title');
+
+    if (saveTitle) {
+      setTitle(saveTitle);
+    }
+  }, []);
+
+  useEffect(() => {
+    localStorage.setItem('title', title);
+  }, [title]);
+  const addNewTodo = (e) => {
+    e.preventDefault();
+
+    const newTodo = {
+      id: prevTodos && prevTodos.length + 1,
```

Comment:
```suggestion
      id: prevTodos.length + 1,
```
I think you don't need the first check. Initially prevTodos.length is 0, so the first id will be 0 + 1 and so on